src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs

```diff
@@ -208,9 +349,75 @@ private static string GenerateUniqueName(IParameterSymbol parameter, ImmutableAr
             throw ExceptionUtilities.Unreachable;
         }
 
-        private async Task<Document> AddSymbolInitializationAsync(
-            Document document, IParameterSymbol parameter, SyntaxNode functionDeclaration, IMethodSymbol method,
-            IBlockOperation blockStatementOpt, ISymbol fieldOrProperty, CancellationToken cancellationToken)
+        private async Task<Document> AddAllSymbolInitializationsAsync(
+            Document document,
+            SyntaxNode constructorDeclaration,
+            IBlockOperation? blockStatementOpt,
+            ImmutableArray<IParameterSymbol> parameters,
+            ImmutableArray<ISymbol> fieldsOrProperties,
+            CancellationToken cancellationToken)
+        {
+            Debug.Assert(parameters.Length >= 2);
+            Debug.Assert(fieldsOrProperties.Length > 0);
+            Debug.Assert(parameters.Length == fieldsOrProperties.Length);
+
+            // Process each param+field/prop in order.  Apply the pair to the document getting the updated document.
+            // Then find all the current data in that updated document and move onto the next pair.
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var nodesToTrack = new List<SyntaxNode> { constructorDeclaration };
+            if (blockStatementOpt != null)
+                nodesToTrack.Add(blockStatementOpt.Syntax);
+
+            var trackedRoot = root.TrackNodes(nodesToTrack);
+            var currentDocument = document.WithSyntaxRoot(trackedRoot);
+
+            for (int i = 0; i < parameters.Length; i++)
```

Comment:
Would this be easier to implement using a SyntaxEditor? Just curious why you chose this approach over it...

Reply:
syntaxeditor turned out to be *much* more difficult. the challenge is with code like:

```
public C(int a, int b, int c, int d, int e)
{
    this.a = a;
    this.d = d;
}
```

So we want to generate, b, c and e, and we need to find and put all of them in the 'right' location respective to both the existing statements *and* the new statements we're adding.  this is where it gets challening.  both 'b' and 'c' come after 'a'.  but if we use the editor and add them one at a time in order we get:

```
this.c = c;
this.b = b;
```

which is reversed.

Processing parameters in reverse order also has issues of it's own.  

So it just was much simpler and saner to just do: add one parameter.  then reconstruct the state, then add the next, and so on.